feature-extensions/apigee/setup.py

```diff
@@ -0,0 +1,123 @@
+    group.add_argument(
+        "-i",
+        "--install",
+        help="Install {0} feature extension".format(featureName),
+        action="store_true")
+    group.add_argument(
+        "-u",
+        "--uninstall",
+        help="Uninstall {0} feature extension".format(featureName),
+        action="store_true")
+    args = parser.parse_args()
+
+    if args.install:
+        install()
+
+    if args.uninstall:
+        uninstall()
+
+
+def install():
+    print(
+        colors.OKGREEN +
+        "\nThis will install {0} functionality into your Jazz deployment.\n".format(featureName)
+        + colors.ENDC)
+    print(
+        colors.OKGREEN +
+        "This installer will use whatever AWS credentials you have configured by running 'aws configure'.\n"
+        + colors.ENDC)
+    print(
+        colors.WARNING +
+        "Please make sure you are using the same AWS credentials you used to install your Jazz deployment\n\n"
+        + colors.ENDC)
+    runTerraform(getRegion(), getAWSAccountID(), getEnvPrefix(), True)
+
+
+def uninstall():
+    print(
+        colors.OKGREEN +
+        "\nThis will remove {0} functionality from your Jazz deployment.\n".format(featureName)
+        + colors.ENDC)
+
+    terraformStateSanityCheck()
+    runTerraform(getRegion(), getAWSAccountID(), getEnvPrefix(), False)
+
+
+def runTerraform(region, accountId, envPrefix, install):
+    print(
+        colors.OKBLUE + 'Initializing and running Terraform.\n' + colors.ENDC)
+    subprocess.check_call(['terraform', 'init'], cwd='./terraform')
+
+    subprocess.check_call(
+        [
+            'terraform', 'apply' if install else 'destroy', '-auto-approve',
+            '-var', 'region={0}'.format(region),
+            '-var', 'jazz_aws_accountid={0}'.format(accountId),
+            '-var', 'env_prefix={0}'.format(envPrefix)
+        ],
+        cwd='./terraform')
+
+
+def terraformStateSanityCheck():
+    print(colors.OKBLUE +
+          'Making sure you have not deleted the Terraform .tfstate file...' +
+          colors.ENDC)
+    if not os.path.isfile('./terraform/terraform.tfstate'):
+        print(colors.FAIL +
+              'Cannot find the Terraform .tfstate file! No uninstall possible'
+              + colors.ENDC)
+
+
+def getRegion():
+    region = raw_input(
+        "Please enter the region where your Jazz installation lives: ")
+
+    if region is "":
+        print("No region entered, defaulting to 'us-east-1'")
+        region = "us-east-1"
+    return region
+
+
+# TODO Obviously it would be better if we could somehow automatically infer the
+# env prefix that the user used to install Jazz proper, but I don't see a
+# robust way to do that yet, so KISS is a better principal to follow here.
+# Also, there are no programmatic side effects if they happen to enter a
+# different prefix.
+def getEnvPrefix():
+    return raw_input(
+        "Please enter the environment prefix you used for your Jazz install: ")
```

Comment:
To be consistent with the terminology which we use during install, update this "stack prefix" or similar.

Reply:
In our main installer we call the variable `envPrefix`, so I went with that name.

Should it be `stackPrefix` instead?

Reply:
The installer text presented to the user says something like `stack prefix`. I would retain that text phrasing to keep it consistent. We can choose a different one, but I wouldn't use environment prefix as it has a certain connotation within Jazz so this can get confusing.  

Doing a quick search in installer, I do see that we were being inconsistent in our variable name for this (envPrefix, stack_prefix, stackPrefix). Let us agree on one and then try to make it consistent as we touch the various pieces of code.